docs/streams/developer-guide/processor-api.html

```diff
@@ -396,6 +397,21 @@ <h2>
                 </ul>
                 </p>
             </div>
+            <div class="section" id="readonly-state-stores">
+                <span id="streams-developer-guide-state-store-readonly"></span><h3><a class="toc-backref" href="#id12">ReadOnly State Stores</a><a class="headerlink" href="#readonly-state-stores" title="Permalink to this headline"></a></h3>
+                <p>The changelog topic for a fault-tolerant state store can also be used to perform lookups against. It is
+                    crucial to understand that there can be only one writer to this state store, but there can be multiple
+                    readers.</p>
+
+                <p>This allows you to use a readonly state store for looking up reference data managed by another processor
+                    or even another process all together. The messages within the compacted topic may be projected (subset of data extracted)
+                    before going into the readonly state store. Therefor we can optimize what the data within our readonly state
+                    store should look like, depending on the use-case it needs to serve.</p>
+
+                <p><b>note:</b> beware of the partitioning requirements when using readonly statestores for lookups during
+                    processing. You might want to make sure the original changelog topic is co-partitioned with the processors
+                    reading the readonly statestore.</p>
```

Comment:
```suggestion
                    reading the read-only state store.</p>
```

Reply:
This reminds me: should we allow to pass in a custom partitioner? 🤔 (Could also be follow up work though...)

Reply:
Eventually this would be needed, but I think it is indeed best to do that as a followup tbh.